scylla-cdc-driver3/src/main/java/com/scylladb/cdc/cql/driver3/Driver3SchemaBuilder.java

```diff
@@ -12,125 +13,477 @@
 import com.datastax.driver.core.TableMetadata;
 import com.datastax.driver.core.TupleType;
 import com.datastax.driver.core.UserType;
+import com.datastax.driver.core.KeyspaceMetadata;
 import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.scylladb.cdc.model.worker.ChangeSchema;
 
-public class Driver3SchemaBuilder {
+/**
+ * A builder of {@link ChangeSchema} based on
+ * the base table schema and CDC log schema.
+ * <p>
+ * The implementation does not require that
+ * the base table schema and CDC log schema
+ * must both be fresh.
+ * <p>
+ * For example, in case of <code>ALTER TABLE</code>,
+ * base table schema containing a schema before
+ * that operation and CDC log schema containing a schema
+ * after that operation, this builder will
```

Comment:
How is it possible for CDC log schema to be after ALTER and base table schema to be before ALTER? I can imagine base table schema being after ALTER and CDC log being before but can't think of a way to achieve the opposite.

Reply:
Did some more real-life testing. Curiously enough, I am able to reproduce CDC log schema after ALTER and base table schema before ALTER, but not the other way around.

Code to reproduce:
```java
package com.datastax.driver.core;

import com.datastax.driver.core.Cluster;
import com.datastax.driver.core.Session;

public class Testing {
    public static void main(String[] args) {
        Cluster c = Cluster.builder().addContactPoint("127.0.0.2").build();
        Session s = c.connect();

        while (true) {
            String baseTableType = c.getMetadata().getKeyspace("ks").getTable("t").getColumn("v").getType().getName().toString();
            String cdcTableType = c.getMetadata().getKeyspace("ks").getTable("t_scylla_cdc_log").getColumn("v").getType().getName().toString();
            if (!baseTableType.equals(cdcTableType)) {
                System.out.println("Base table is: " + baseTableType);
                System.out.println("CDC table is: " + cdcTableType);
                s.close();
                c.close();
                return;
            }
        }
    }
}
```

Before running: `CREATE TABLE ks.t(pk int, ck int, v int, PRIMARY KEY(pk, ck)) WITH cdc = {'enabled': true};`
While running: `ALTER TABLE ks.t ALTER v TYPE blob;`
Quite frequently you will see:
```
Base table is: int
CDC table is: blob
```
(CDC after ALTER, base before ALTER)

Right now I am having a difficulty reproducing the other way around. Since this is such a tricky thing, which could change with innocuous change in driver/Scylla, I think it is better to prepare for both cases.

Reply:
However, in actual code (this PR), the situation is a bit different (chronologically):

1. We prepare the statement on CDC log table.
2. We get the schema from the PreparedStatement on CDC log (first row returned by PreparedStatement).
3. We get the schema of base table from `Cluster`.

If `ALTER` happens before 1, CDC log schema will be fresh, but base table schema could be stale (generally there is 
at least ~1s delay in schema in `Cluster`). If `ALTER` happens after 1, CDC schema will be old, but maybe there was enough time for `Cluster` to acknowledge new schema, so the base table schema will be fresh.

I will consider updating a comment in the source code with this explanation above.

Reply:
I see.